### PR TITLE
Potential Slack Rate Limit Fix

### PR DESCRIPTION
**Pull request by Linuturk**
So, it seems like the list all options might be hitting a slack rate limit. I've modified the list all function to send one bulk message instead of flooding the channel with a bunch of single messages. Version bumped and documented the list all command as well.

### CODE REVIEW

**Comment by Linuturk** [comment]
fixes #11

File: src/circle-ci.coffee [hunk 1]
```diff
@@ -81,11 +81,15 @@ listProjectsByStatus = (msg, projects, status) ->
...
-      msg.send "Projects where the last build's status is #{status}:"
+      message = "Projects where the last build's status is #{status}:\n"
       for project in projects
         build_branch = project.branches[project.default_branch]
         last_build = build_branch.recent_builds[0]
-        msg.send "#{toDisplay(last_build.outcome)} in build https://circleci.com/gh/#{project.username}/#{project.reponame}/#{last_build.build_num} of #{project.vcs_url} [#{project.default_branch}]"
+        message = message + "#{toDisplay(last_build.outcome)} in build https://circleci.com/gh/#{project.username}/#{project.reponame}/#{last_build.build_num} of #{project.vcs_url} [#{project.default_branch}]\n"
+      sendBulkMessage(msg, message)
+
+sendBulkMessage = (msg, message) ->
```
Comment (dylanlingelbach) [inline src/circle-ci.coffee]: @Linuturk - Why have this function rather than just calling `msg.send`?

**Comment by dylanlingelbach** [comment]
@Linuturk - sorry for the delay on this, the last two weeks have been busy for me.

Should this really be fixed in the Slack adapter rather than each plugin that writes messages?

**Comment by Linuturk** [comment]
Well, it has to do with the frequency we are sending many small messages. I wouldn't expect the slack adapter to buffer and dump my messages all at once for me. I'd expect it to immediately send whatever I ask it to send. That means we just have to be smart about the number of messages we are sending at a time.

**Comment by dylanlingelbach** [comment]
Ok, I had the opposite reaction.  I was thinking the Slack adapter would buffer and send individual messages.  The reason I thought this was the Slack adapter is the only place that understands the Slack rate limits and how best to handle them.  Otherwise every Hubot plugin that sends many small messages will have to write code to handle the Slack rate limiter.  And if another adapter barfs on large messages it would be impossible to support both.  Does that make sense?

I'm not saying no to this, just trying to discuss the possibilities.

**Comment by Linuturk** [comment]
That's a good point as well. The rate limits aren't something they publish, apparently on purpose. Maybe it needs better error handling with more intelligent error messages on the Slack side, as the existing error returned breaks our JSON parsing something awful and made this hard to track down.

Ultimately, I don't really want to wait for them to implement a fix on their side if we can work around here and play the friendly neighbor. I don't suppose it would hurt to drop an issue on this in their tracker, but I'd also like to make sure I'm not overlooking another problem we have. I'm not 100% on this code base, and I might be fixing a symptom rather than the real problem. I think I've gotten lucky so far that my contributions work the way I expect, but those are built on the existing foundation which is still hairy for me.

**Comment by dylanlingelbach** [comment]
Yeah, makes total sense.  Would you mind opening an issue on their tracker?  We use HipChat, not Slack so I won't be of much help if they have questions.

Could you link back to this issue as well?

I have no problem hacking something in so we can move forward until they fix, but would like to have an issue that I can track so can unwind this if fixed.

**Comment by Linuturk** [comment]
There we go :+1:

**Comment by Linuturk** [comment]
looks like this feel off the radar and didn't get merged before the other PR

**Comment by dylanlingelbach** [comment]
@Linuturk - sorry I wasn't clear about this.  I would like to wait to merge this in until a resolution is reached on slackhq/hubot-slack.  I really think the right place to fix is is in the slack adapter.

Is running off your fork until then a big issue for you?

**Comment by Linuturk** [comment]
Not a big issue. We are running from the official release, not a fork.

**Comment by Linuturk** [comment]
@dylanlingelbach We aren't seeing any love from the slack guys. Does it make sense at this point to just work around it?

**Comment by dylanlingelbach** [comment]
@Linuturk - yup, I think it does. Do you mind dealing with these merge conflicts? And then I'll update and cut the release. Just started at a new company today so might take me a day or two to turn it around.

**Comment by Linuturk** [comment]
@dylanlingelbach that should be good :)

**Pull request merged**